libs/features/personalization/personalization.js

```diff
@@ -327,70 +344,73 @@ function registerInBlockActions(cmd, manifestId, targetManifestId) {
   config.mep.inBlock[blockName].commands.push(command);
 }
 
-function getSelectedElement(selector, action, rootEl) {
-  if (!selector) return null;
-  if ((action.includes('appendtosection') || action.includes('prependtosection'))) {
-    if (!selector.includes('section')) return null;
-    const section = selector.trim().replace('section', '');
-    if (section !== '' && Number.isNaN(section)) return null;
-  }
-  if (checkSelectorType(selector) === 'fragment') {
+function modifySelectorTerm(term) {
+  const specificSelectors = {
+    section: 'main > div',
+    'primary-cta': 'p strong a',
+    'secondary-cta': 'p em a',
+    'action-area': 'p:has(em a, strong a)',
+  };
+  const otherSelectors = ['row', 'col'];
+  const htmlEls = ['div', 'a', 'p', 'strong', 'em', 'picture', 'source', 'img', 'h'];
+  let startText = term.match(/^[a-zA-Z/./-]*/)[0]?.toLowerCase();
+  const endNumber = term.match(/[0-9]*$/)[0];
+  if (!startText || htmlEls.includes(startText)) return term;
+  let modifiedTerm = term;
+  if (otherSelectors.includes(startText) || Object.keys(specificSelectors).includes(startText)) {
+    if (otherSelectors.includes(startText)) modifiedTerm = modifiedTerm.replace(startText, '> div');
+    else modifiedTerm = modifiedTerm.replace(startText, specificSelectors[startText]);
+    if (endNumber) modifiedTerm = modifiedTerm.replace(endNumber, `:nth-child(${endNumber})`);
+    return modifiedTerm;
+  }
+  if (!startText.startsWith('.')) {
+    startText = `.${startText}`;
+    modifiedTerm = `.${modifiedTerm}`;
+  }
+  if (endNumber) {
+    modifiedTerm = modifiedTerm.replace(endNumber, `:nth-child(${endNumber} of ${startText})`);
+  }
+  return modifiedTerm;
+}
+export function modifyNonFragmentSelector(selector) {
+  let modifiedSelector = selector;
+  const terms = modifiedSelector.split('>').join(' > ').split(',').join(' , ')
+    .split(/\s+/);
+  terms.forEach((term, i) => {
+    if (term.toLowerCase() === 'main') {
+      const next = terms[i + 1]?.toLowerCase();
+      const nextNext = terms[i + 2]?.toLowerCase();
+      if (next?.startsWith('section') || nextNext?.startsWith('section')) {
+        terms[i] = '';
+        if (next === '>') terms[i + 1] = '';
+      }
+    } else {
+      terms[i] = modifySelectorTerm(term);
+    }
+  });
+  modifiedSelector = terms.join(' ');
+  return modifiedSelector;
+}
```

Comment:
Seems like this is to remove `main sectionX` or `main > sectionX`?  This might be a case where regexp is better?

```suggestion
export function modifyNonFragmentSelector(selector) {
  return selector
    .split('>').join(' > ')
    .split(',').join(' , ')
    .replace(/^main( > section\d*| section\d*)/, '')
    .split(/\s+/)
    .map(modifySelectorTerm)
    .join(' ');
}
```

Reply:
yeah, this is better